test/41-device-api-v1.spec.ts

```diff
@@ -973,9 +973,10 @@ describe('SupervisorAPI [V1 Endpoints]', () => {
 						expect(await exists(noProxyPath)).to.be.false;
 					});
 
-				expect(restartServiceSpy.callCount).to.equal(2);
-				expect(restartServiceSpy.args).to.deep.equal([
-					['resin-proxy-config'],
+				// Should query service(s) to see if a restart is required
+				expect(servicePartOfSpy.callCount).to.equal(2);
```

Comment:
I think the code is fine, but the tests you modified now don't test the `restartService` branch of the code. Could you add a test to check that that the function `restartService` is called when the `servicePartOf` throws an exception? 

I know the implementation is not great, because to test we need to know the internal details of the method, but for the time being is probably better to add that extra test case.